File: app/src/main/java/com/nawin/androidmvparchitecture/auth/login/LoginContract.java
package com.nawin.androidmvparchitecture.auth.login;

import com.nawin.androidmvparchitecture.BasePresenter;
import com.nawin.androidmvparchitecture.BaseView;
import com.nawin.androidmvparchitecture.data.model.viewmodel.LoginViewModel;

/**
 * Created by nawin on 6/13/17.
 */

interface LoginContract {

    interface View extends BaseView<Presenter> {
        void showLoginProgress();

        void showLoginSuccess(LoginViewModel loginViewModel);

        void showLoginError(String message);

        void showNetworkNotAvailableError();
    }

    interface Presenter extends BasePresenter {
        void onLogin(String username, String password);
    }
}

File: app/src/main/java/com/nawin/androidmvparchitecture/auth/login/LoginContract.kt
package com.nawin.androidmvparchitecture.auth.login

import com.nawin.androidmvparchitecture.BasePresenter
import com.nawin.androidmvparchitecture.BaseView

/**
 * Created on 12/13/17.
 */
interface LoginContract {

    interface View : BaseView<Presenter> {
        fun showLoginProgress()

        fun showLoginSuccess()

        fun showLoginError(message: String)

        fun showNetworkNotAvailableError()
    }

    interface Presenter : BasePresenter {
        fun onLogin(username: String, password: String)
    }
}
